api/src/env.ts

```diff
@@ -380,37 +382,37 @@ export const getEnv = () => env;
  * When changes have been made during runtime, like in the CLI, we can refresh the env object with
  * the newly created variables
  */
-export function refreshEnv(): void {
+export async function refreshEnv(): Promise<void> {
 	env = {
 		...defaults,
 		...process.env,
-		...processConfiguration(),
+		...(await processConfiguration()),
 	};
 
 	process.env = env;
 
 	env = processValues(env);
 }
 
-function processConfiguration() {
+async function processConfiguration() {
 	const configPath = path.resolve(process.env['CONFIG_PATH'] || defaults['CONFIG_PATH']);
 
 	if (fs.existsSync(configPath) === false) return {};
 
 	const fileExt = path.extname(configPath).toLowerCase();
 
-	if (fileExt === '.js') {
-		const module = require(configPath);
-		const exported = module.default || module;
+	if (['.js', '.cjs', '.mjs'].includes(fileExt)) {
```

Comment:
Very side tangent here, but should we use this extensions constant over here?

https://github.com/directus/directus/blob/0c56c46fdf8dfd0a40ae04b3ec6270c42edf1077/packages/constants/src/files.ts#L1

this is a rather random question, shouldn't really block this PR at all as it works as is 👍

Reply:
Nice 🚀❤️